src/cunumeric/unary/scalar_unary_red_template.inl

```diff
@@ -28,8 +28,13 @@ namespace cunumeric {
 
 using namespace legate;
 
-template <VariantKind KIND, UnaryRedCode OP_CODE, Type::Code CODE, int DIM>
+template <VariantKind KIND, UnaryRedCode OP_CODE, Type::Code CODE, int DIM, bool HAS_WHERE>
 struct ScalarUnaryRed {
+  ScalarUnaryRed(ScalarUnaryRedArgs& args) { assert(false); }
```

Comment:
What is this trying to do? Bar the use of the general template? Or bar a non-const copy? If the point is to deny the use of the general template, prefer doing something like
```c++
template <typename T>
struct Foo {
  // Note tautological expression still dependent on template parameter
  static_assert(!std::is_same_v<T, T>);
};
```
If you'd like to instead bar the use of copy ctor, prefer the usual `= delete`:
```c++
template <typename T>
struct Foo {
  Foo(const Foo&) = delete;
};
```

Reply:
Thank you, @Jacobfaib . This is not necessary anymore. the purpose for this was to "prevent the use of general template"